doc/development/contribution/style.rst

```diff
@@ -20,8 +25,62 @@ Coding style and quality
   test suite are in the style of Python's unit test module. If you extend those files, you might continue in this style,
   but please use ``pytest`` style for any new test files.
 
-* Please keep the first line of your commit messages short. When referencing an issue, please phrase it like
-  ``Fix #123 -- Problems with order creation`` or ``Refs #123 -- Fix this part of that bug``.
+Commits and Pull Requests
+-------------------------
+
+
+
+Most commits should start as pull requests, therefore this applies to the titles of pull requests as well since
+the pull request title will become the commit message on merge. We prefer merging with GitHub's "Squash and merge"
+feature if the PR contains multiple commits that do not carry value to keep. If there is value in keeping the
+individual commits, we use "Rebase and merge" instead. Merge commits should be avoided.
+
+* The commit message should start with a single subject line and can optionally followed by a commit message body.
+
+  * The subject line should be the shortest possible representation of what the commit changes. Someone who reviewed
+    the commit a couple of weeks ago should able to immediately remember the commit based on the subject line and tell
```

Comment:
```suggestion
  * The subject line should be the shortest possible representation of what the commit changes. Someone who reviews
    the commit in a couple of weeks should able to immediately remember the commit based on the subject line and tell
```

Reply:
Hmm, that's not precisely what I meant. My sentiment was: If I'm the one reviewing the PR, and I'm the one writing the release notes four weeks later, I want the subject line to be all I need to recognize and remember the change.

Reply:
Ah, now I get it! Maybe this way around makes it clearer?

```suggestion
  * The subject line should be the shortest possible representation of what the commit changes. Someone who reviewed
    the commit should able to immediately remember the commit in a couple of weeks based on the subject line and tell
```